pybedlite/overlap_detector.py

```diff
@@ -258,70 +339,101 @@ def overlaps_any(self, interval: Interval) -> bool:
 
         Returns:
             The list of intervals in this detector that overlap the given interval, or the empty
-            list if no overlaps exist.  The intervals will be return in ascending genomic order.
+            list if no overlaps exist. The intervals will be returned sorted using the following
+            sort keys:
+
+                * The interval's start
+                * The interval's end
+                * The interval's strand (if defined)
+                * The interval's reference sequence name
         """
         tree = self._refname_to_tree.get(interval.refname)
         if tree is None:
             return []
         else:
             if not self._refname_to_indexed[interval.refname]:
                 tree.index()
-            ref_intervals: List[Interval] = self._refname_to_intervals[interval.refname]
+            ref_intervals: List[GenericGenomicSpan] = self._refname_to_intervals[interval.refname]
             # NB: only return unique instances of intervals
-            intervals: Set[Interval] = {
+            intervals: Set[GenericGenomicSpan] = {
                 ref_intervals[index]
                 for _, _, index in tree.overlap(interval.refname, interval.start, interval.end)
             }
             return sorted(
-                intervals, key=lambda intv: (intv.start, intv.end, intv.negative, intv.name)
+                intervals,
+                key=lambda intv: (
+                    intv.start,
+                    intv.end,
+                    self._negative(intv),
+                    intv.refname,
+                ),
             )
 
-    def get_enclosing_intervals(self, interval: Interval) -> List[Interval]:
+    @staticmethod
+    def _negative(interval: GenomicSpan) -> bool:
+        """
+        True if the interval is negatively stranded, False if the interval is unstranded or
+        positively stranded.
+        """
+        return getattr(interval, "negative", False)
+
+    def get_enclosing_intervals(self, interval: GenomicSpan) -> List[GenericGenomicSpan]:
         """Returns the set of intervals in this detector that wholly enclose the query interval.
-        i.e. query.start >= target.start and query.end <= target.end.
+        i.e. `query.start >= target.start` and `query.end <= target.end`.
+
+        Args:
+            interval: the query interval
 
-          Args:
-              interval: the query interval
-          Returns:
-              The list of intervals in this  detector that enclose the query interval.
-              The intervals will be returned in ascending genomic order.
+        Returns:
+            The list of intervals in this detector that enclose the query interval. The intervals
+            will be returned sorted using the following sort keys:
+
+                * The interval's start
+                * The interval's end
+                * The interval's strand (if defined)
+                * The interval's reference sequence name
```

Comment:
Do you think these clarifications help?  If so, can you add them everywhere?
```suggestion
                * The interval's start, ascending
                * The interval's end, ascending
                * The interval's strand, positive and negative (assumed to be positive if not defined)
                * The interval's reference sequence name (lexicographically)
```